### PR TITLE
Add append to file if the cert and key filenames are same

### PR DESCRIPTION
**Pull request by trilokgm**
**What this PR does?**

When spiffe-helper is configured with same filename for key and certificate, append to single file is made possible.

**Fixes**
#7

### CODE REVIEW

File: spiffe-helper.go [hunk 1]
```diff
@@ -209,6 +219,28 @@ func (s *sidecar) writeKey(file string, data []byte) error {
...
+	}
+
+	f, err := os.OpenFile(file, os.O_WRONLY|os.O_CREATE|os.O_APPEND, keyFileMode)
+	if err != nil {
+		return err
+	}
+
+	_, err = f.Write(pem.EncodeToMemory(b))
+	if err != nil {
+		return err
```
Comment (maxlambrecht) [inline spiffe-helper.go]: err = f.Close() is needed before returning here.

File: spiffe-helper.go [hunk 2]
```diff
@@ -209,6 +219,28 @@ func (s *sidecar) writeKey(file string, data []byte) error {
 	return ioutil.WriteFile(file, pem.EncodeToMemory(b), keyFileMode)
 }
 
+// writeKey takes a private key as a slice of bytes,
```
Comment (maxlambrecht) [inline spiffe-helper.go]: amend the comment

File: spiffe-helper.go (hunk 1 above)
Comment (maxlambrecht) [reply spiffe-helper.go]: `writeKey` and `appendKey` could be refactored to avoid some duplicated code

File: spiffe-helper.go (hunk 2 above)
Comment (trilokgm) [reply spiffe-helper.go]: Changed the implementation of write key to accomodate append parameters for open file and changed the implementation to use write on *File.

File: spiffe-helper.go (hunk 1 above)
Comment (trilokgm) [reply spiffe-helper.go]: refactored writeKey implementation to accomodate append mode.

**Pull request merged**